appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockBarChartDataModel.java

```diff
@@ -0,0 +1,200 @@
+package com.google.appinventor.client.editor.simple.components;
+
+import com.google.appinventor.client.output.OdeLog;
+import org.pepstock.charba.client.data.BarDataset;
+import org.pepstock.charba.client.data.Data;
+import org.pepstock.charba.client.data.DataPoint;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Chart Data Model for Bar Chart based views.
+ */
+public class MockBarChartDataModel extends MockChartDataModel<BarDataset> {
+  // Keep track of the associated Mock Bar Chart View to be able
+  // to invoke axis label updating.
+  private MockBarChartView view;
+
+  /**
+   * Creates a new Mock Bar Chart Data Model object instance,
+   * linking it with the specified MockBarChartView object.
+   * @param view  MockBarChartView to link the model to
+   */
+  public MockBarChartDataModel(MockBarChartView view) {
+    super(view.getChartWidget().getData());
+
+    // Set local view variable (needed to invoke the updateLabels method)
+    this.view = view;
```

Comment:
@ewpatton Very good point; I noticed actually that passing in the View makes more sense (as I did in the MockBarChartDataModel) and this would also allow setting the parameter.